store/gcworker/gc_worker.go

```diff
@@ -1058,17 +1083,90 @@ func (w *GCWorker) legacyResolveLocks(ctx context.Context, safePoint uint64, con
 	logutil.Logger(ctx).Info("[gc worker] finish resolve locks",
 		zap.String("uuid", w.uuid),
 		zap.Uint64("safePoint", safePoint),
+		zap.Uint64("try-resolve-locks-ts", tryResolveLocksTS),
 		zap.Int("regions", runner.CompletedRegions()))
 	metrics.GCHistogram.WithLabelValues("resolve_locks").Observe(time.Since(startTime).Seconds())
 	return nil
 }
 
-func (w *GCWorker) resolveLocksForRange(ctx context.Context, safePoint uint64, startKey []byte, endKey []byte) (rangetask.TaskStat, error) {
+// getTryResolveLocksTS gets the TryResolveLocksTS
+// that is defined as `now() - gcTryResolveLocksIntervalFromNow`.
+func (w *GCWorker) getTryResolveLocksTS() (uint64, error) {
+	now, err := w.getOracleTime()
+	if err != nil {
+		return 0, err
+	}
+
+	tryResolveLocksTime := now.Add(-gcTryResolveLocksIntervalFromNow)
+	return oracle.GoTimeToTS(tryResolveLocksTime), nil
+}
+
+// batchResolveExpiredLocks tries to resolve expired locks with batch method.
+// Travesal the given locks and check that:
+// 1. If the ts of lock is equal with or smaller than forceResolveLocksTS(acually equals safepoint),
+// it will rollback the txn, no matter the lock is expired of not.
+// 2. If the ts of lock is larger than forceResolveLocksTS, it will check status of the txn.
+// Resolve the lock if txn is expired, Or do nothing.
+func (w *GCWorker) batchResolveExpiredLocks(
+	bo *tikv.Backoffer,
+	locks []*txnlock.Lock,
+	loc tikv.RegionVerID,
+	forceResolveLocksTS uint64,
+	tryResolveLocksTS uint64,
+) (bool, error) {
+	if len(locks) == 0 {
+		return true, nil
+	}
+
+	forceResolveLocks := make([]*txnlock.Lock, 0, len(locks))
+	tryResolveLocks := make([]*txnlock.Lock, 0, len(locks))
+	for _, l := range locks {
+		if l.TxnID <= forceResolveLocksTS {
+			forceResolveLocks = append(forceResolveLocks, l)
+		} else {
+			tryResolveLocks = append(tryResolveLocks, l)
+		}
+	}
+
+	logutil.BgLogger().Info("BatchResolveLegacyLocks",
```

Comment:
Don't use "legacy", it may confuse people with other things. I think it's ok to change it to be same with the function name `BatchResolveExpiredLocks`